src/main/java/seedu/dietbook/command/CalculateCommand.java

```diff
@@ -44,6 +44,7 @@ public void execute(Manager manager, Ui ui) throws DietException {
             if (processedParam.length == 1) {
                 ui.printAllIntake(this.calorie, this.carb, this.protein, this.fat);
             } else if (processedParam.length == 2) {
+                InputChecker.checkDateValidity(processedParam[1]);
```

Comment:
Perhaps consider refactoring to avoid long methods if time permits. Similarly, consider this for other long methods.